File: 006/Problem6.java
public class Problem6 {
    public static void main(String[] args) {
        final long startTime = System.nanoTime();
        for (int i = 0; i < 1000; i++) {
            runProblem();
        }
        final long endTime = System.nanoTime();
        System.out.println("Function took " + (endTime - startTime));
    }

    public static int SumOfSquares(int num) {
        int sum = 0;
        for (int i = 0; i < num + 1; i++) {
            sum += i * i;
        }
        return sum;
    }

    public static int SquareOfSum(int num) {
        int sum = 0;
        for (int i = 0; i < num + 1; i++) {
            sum += i;
        }
        return sum * sum;
    }

    static void runProblem() {
        System.out.println(SquareOfSum(100) - SumOfSquares(100));
    }
}

File: 006/Problem6.kt
import kotlin.system.measureNanoTime

fun main() {
    val loops = 10000
    val timeInNanos = measureNanoTime {
        for (i in 1..loops) {
            problem()
        }
    }
    println("Function Took $timeInNanos Nano seconds for $loops loops")
}

fun sumOfSquares(num: Int): Int {
    var sum = 0
    for (n in 1..num) {
        sum += n * n
    }
    return sum
}

fun squareOfSum(num: Int): Int {
    var sum = 0
    for (n in 1..num) {
        sum += n
    }
    return sum * sum
}

fun problem() {
    val naturalNum = 100
    println(squareOfSum(naturalNum) - sumOfSquares(naturalNum))
}